### PR TITLE
ICMSLST-2930 Refactor Template model to use TemplateCodes enum.

### PR DESCRIPTION
**Pull request by MattHolmes123**

### CODE REVIEW

File: web/mail/emails.py [hunk 1]
```diff
@@ -501,12 +502,12 @@ def send_case_email(case_email: CaseEmailModel, sent_by: User) -> None:
...
 def create_case_email(
     application: ImpOrExp,
-    email_type: EmailTypes,
+    email_type: TemplateCodes,
     to: str | None = None,
     cc: list[str] | None = None,
     attachments: QuerySet | None = None,
 ) -> CaseEmailModel:
-    template_code = email_type.value
+    template_code = email_type
```
Comment (marcuspp) [inline web/mail/emails.py:510]: The CaseEmail model has choices as `CASE_EMAIL_TYPES` for this field

**Review by marcuspp — CHANGES_REQUESTED** [review]
I think anywhere where you are creating a case email the email type should be a `CaseEmailCode`

File: web/mail/emails.py [hunk 2]
```diff
@@ -501,12 +502,12 @@ def send_case_email(case_email: CaseEmailModel, sent_by: User) -> None:
 
 def create_case_email(
     application: ImpOrExp,
-    email_type: EmailTypes,
+    email_type: TemplateCodes,
```
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/mail/emails.py:505]: Yer I agree (I nearly changed it to this before I submitted the PR).
I added TemplateCodes quite late in the PR and I'd already typed create_case_email.

[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/mail/emails.py:505]: Actually no I've remembered why I left it as is.

[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/mail/emails.py:505]: I've rememered why I left it as I did.
image
create_case_email calls `get_email_template_subject_body` to fetch the template and that is correctly typed like this:

image

**Review by marcuspp — APPROVED** [review]

**Pull request merged**